### PR TITLE
Add pre commit hooks for formatting

### PR DESCRIPTION
**Pull request by liu-allan**
Made to address #1206, black and isort are run on the staged files

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -111,6 +111,7 @@ pexpect==4.8.0
 pickleshare==0.7.5
 Pillow==8.2.0
 pluggy==1.0.0
+pre-commit==2.16.0
```
Comment (Adaickalavan) [inline requirements.txt:114]: The `requirements.txt` file will be overwritten and updated automatically, on commit to the develop branch, by `ci-dependencies-linux.yml` in the CI.

Currently, we only include [test,train,camera-obs] options from `setup.py` inside `requirements.txt` file.

If needed, perhaps `pre-commit==2.16.0` and `pylint==2.12.2` can be included in the [dev] option in `setup.py` file.

Comment (liu-allan) [reply requirements.txt]: I have made this change, is there anything else?

File: setup.py [hunk 2]
```diff
@@ -80,6 +81,8 @@
             "black==20.8b1",
             "grpcio-tools==1.32.0",
             "isort==5.7.0",
+            "pre-commit==2.16.0",
+            "pylint==2.12.2",
```
Comment (Gamenot) [inline setup.py:85]: Is there some description here of what we would need to do to get `pre-commit` set up?

**Review by Gamenot — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (liu-allan) [reply setup.py:85]: user just has to run `pre-commit install`, where should i include this instruction?

Comment (Gamenot) [reply setup.py:85]: I think add instructions for setup in in the `CONTRIBUTING.md`.

Comment (Gamenot) [reply setup.py:85]: You will need to take a read-through and update the file.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -18,10 +18,11 @@ We encourage all forms of contributions to SMARTS, not limited to:
 
 ## Setting up your dev environment
 
-In addition to following the setup steps in the README, you'll need to install the [dev] dependencies.
+In addition to following the setup steps in the README, you'll need to install the [dev] dependencies and install the pre-commit hooks.
 
 ```bash
 pip install -e .[dev]
+pre-commit install
```
Comment (Gamenot) [inline CONTRIBUTING.md:25]: This should not be needed because it will be installed with `[dev]`.

Comment (liu-allan) [reply CONTRIBUTING.md:25]: ah with that i meant the `pre-commit install` command

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -90,13 +90,9 @@ Tag the issue as a feature request using `enhancement` and if it takes more than
 
 ### Python(Format)
 
-1. Always run `make format` before committing.
-
```
Comment (Gamenot) [inline CONTRIBUTING.md:94]: nit: Perhaps reference the need to have `pre-commit` set up for the repository.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -90,13 +90,9 @@ Tag the issue as a feature request using `enhancement` and if it takes more than
...
 The project follows a strict format requirement for python code. We made a decision early on in the project to use [Black](https://github.com/psf/black). This makes formatting consistent while eliminating [bike shedding](http://bikeshed.com/).
 If you do not already have it please install it via `pip install black`.
 
-Formatting guarantees that your code will pass the CI formatting test case.
-
```
Comment (Gamenot) [inline CONTRIBUTING.md:99]: I do not think these lines need to be deleted.

File: CONTRIBUTING.md (hunk 3 above)
Comment (liu-allan) [reply CONTRIBUTING.md:25]: it is, this is to setup our pre-commit hooks for use not the actual install

Comment (Gamenot) [reply CONTRIBUTING.md:25]: Oh, sorry, yes!

**Review by Gamenot — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+repos:
+-   repo: https://github.com/psf/black
+    rev: 19.3b0
```
Comment (Adaickalavan) [inline .pre-commit-config.yaml:3]: We are using `black==20.8b1` in `setup.py`

https://github.com/huawei-noah/SMARTS/blob/ce1061bba9e7e9f00510443b2ce09a07d9f6ed64/setup.py#L81

Should we use the same black version here?

**Review by Adaickalavan — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -42,8 +43,8 @@ Please take care in using good commit messages as they're useful for debugging,
 
 ## Pre-Push Checklist
 
-1. Do your best to see that your code compiles locally.
-2. Run `make format`. See [Formatting](#Formatting).
+1. Make sure the pre-commit hooks are installed `pre-commit install`
```
Comment (Adaickalavan) [inline CONTRIBUTING.md:46]:
```suggestion
1. Make sure the pre-commit hooks are installed via `pre-commit install`.
```

**Comment by Gamenot** [comment]
It is not good that we are seeing determinism issues.

**Pull request merged**